pool/pool.go

```diff
@@ -978,25 +922,31 @@ func (c *clientWrapper) incRequests(elapsed time.Duration, method MethodIndex) {
 	}
 }
 
-func (c *clientStatusMonitor) handleError(st apistatus.Status, err error) error {
-	if err != nil {
-		// non-status logic error that could be returned
-		// from the SDK client; should not be considered
-		// as a connection error
-		var siErr *object.SplitInfoError
-		if !errors.As(err, &siErr) {
-			c.incErrorRate()
-		}
-
-		return err
+func (c *clientStatusMonitor) handleError(err error) error {
```

Comment:
Do you need to `return` anything from it now?

Reply:
Actually no, even more we may rename it to something `updateErrorRate`, because this is single responsibility of this function.
Should I refactor this function?

Reply:
Sure, it'd clarify the purpose and simplify the usage.

Reply:
refactored